src/jit/compiler.cpp

```diff
@@ -640,8 +640,15 @@ var_types Compiler::getPrimitiveTypeForStruct(unsigned structSize, CORINFO_CLASS
             // For ARM_SOFTFP, HFA is unsupported so we need to check in another way
             // This matters only for size-4 struct cause bigger structs would be processed with RetBuf
             if (isSingleFloat32Struct(clsHnd))
-#else  // !ARM_SOFTFP
-            if (IsHfa(clsHnd))
+#else // !ARM_SOFTFP
+            if (IsHfa(clsHnd)
+#if !defined(_TARGET_UNIX_) && defined(_TARGET_ARM64_)
```

Comment:
It might be better to define
```
#if defined(_TARGET_ARM64_) && !defined(_TARGET_UNIX_)
#define IGNORE_HFA_FOR_VARARG
```
Don't we have a TARGET_WINDOWS or WIN64?

Reply:
I believe there is only a WINOWS_AMD64_ABI define.

Reply:
I may be missing something.